PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRunner.java

```diff
@@ -163,6 +160,16 @@ public ISparkBatchJob buildSparkBatchJob(@NotNull SparkSubmitModel submitModel,
                     throw new ExecutionException("Error getting access token based on the given ADLS root path");
                 }
 
+                break;
+            case ADLS_GEN2:
+                destinationRootPath = submitModel.getJobUploadStorageModel().getUploadPath();
+                httpObservable = new ADLSGen2Observable();
+                accessToken = submitModel.getJobUploadStorageModel().getAccessKey();
+                if(StringUtils.isBlank(accessToken)){
+                    throw new ExecutionException("Invalid access key input.");
+                }
+
+                jobDeploy = new ADLSGen2Deploy(clusterDetail, httpObservable, accessToken);
```

Comment:
If the accessKey is put into httpObservable, needn't to put it into `ADLSGen2Deploy`